Fixed the review bugs and emoji overlay placement bug

manojmalik20 opened the pull request:
Added the functionality to open custom status modal by clicking the custom status emoji in the sidebar header
Fixed the positioning of the custom status in the status dropdown
Fixed the emoji overlay positioning bug in custom status modal when zooming in more than 110% or in small resolutions

<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
<!--
A description of what this pull request does.
-->

#### Ticket Link
<!--
If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.

  Fixes https://github.com/mattermost/mattermost-server/issues/XXXXX

Otherwise, link the JIRA ticket.
-->

#### Related Pull Requests
<!--
List all PRs related to resolving a ticket. For instance, if you submitted a PR to `mattermost/mattermost-server`, please include it here.
-->
- Has server changes (please link here)
- Has redux changes (please link here)
- Has mobile changes (please link here)

#### Screenshots
<!--
If the PR includes UI changes, include screenshots/GIFs.
-->

manojmalik20 committed:
Fixed the review bugs and emoji overlay placement bug
Added the functionality to open custom status modal by clicking the custom status emoji in the sidebar header
Fixed the positioning of the custom status in the status dropdown
Fixed the emoji overlay positioning bug in custom status modal when zooming in more than 110% or in small resolutions

manojmalik20 committed:
Fixed the type of the defaultHorizontalPosition in the popOverOverlay util

chetanyakan commented on components/custom_status/custom_status_emoji.tsx:
```diff
@@ -36,6 +37,7 @@ const CustomStatusEmoji = (props: ComponentProps) => {
             emoji={customStatus.emoji}
             size={emojiSize}
             emojiStyle={emojiStyle}
+            openModalOnClick={openModalOnClick}
```
Please make this a generic `onClick` method instead.

chetanyakan commented on components/emoji/render_emoji.tsx:
```diff
@@ -1,32 +1,54 @@
     }
     const emojiImageUrl = getEmojiImageUrl(emojiFromMap);
+
+    const openCustomStatusModal = (event: React.MouseEvent<HTMLSpanElement, MouseEvent>) => {
+        if (openModalOnClick) {
+            event.stopPropagation();
+            const customStatusInputModalData = {
+                ModalId: ModalIdentifiers.CUSTOM_STATUS,
+                dialogType: CustomStatusModal,
+                dialogProps: {userId: currentUserId},
+            };
+
+            dispatch(openModal(customStatusInputModalData));
+        }
+    };
```
Please pass this function as an `onClick` method from the component that wants to open the custom status modal.
nit - you can also use an action in `mattermost-webapp` for the modal data.

chetanyakan commented on components/emoji_picker/emoji_picker_tabs.jsx:
```diff
@@ -70,6 +70,13 @@ export default class EmojiPickerTabs extends PureComponent {
+                // Placement right is only for the emoji overlay in the custom status modal and this styling is specific for that
+                pickerStyle = {
+                    ...this.props.style,
+                    left: this.props.style.left + 3,
+                    top: this.props.style.top + 3,
```
Why don't we pass the correct `top` and `left` values instead of applying a correction here?

chetanyakan commented on components/legacy_sidebar/header/sidebar_header_dropdown_button.jsx:
```diff
@@ -79,6 +79,7 @@ export default class SidebarHeaderDropdownButton extends React.PureComponent {
                                 verticalAlign: 'top',
                                 marginLeft: 4,
                             }}
+                            openModalOnClick={true}
```
Use the aforementioned suggestions.

Define the implementation to open custom status modal and pass a generic `onClick` method here calling that function.

manojmalik20 committed:
Fixed review comments
Changed the openModalOnClick functionality to take generic onClick as prop in the CustomStatusEmoji
Modified the emoji picker overlay to take props to handle right position of the overlay

chetanyakan commented on components/emoji_picker/emoji_picker_tabs.jsx:
```diff
@@ -70,20 +72,16 @@ export default class EmojiPickerTabs extends PureComponent {
                     bottom: this.props.style.bottom,
                     right: this.props.rightOffset,
                 };
```
Can we set `this.props.style.top` and `style.left` here instead of using `props.topOffset` and `props.leftOffset`

manojmalik20 replied:
No, we cannot set because the style object in the props because it is injected by the react-bootstrap Overlay component

chetanyakan commented on components/legacy_sidebar/header/sidebar_header_dropdown_button.jsx:
```diff
@@ -79,7 +81,15 @@ export default class SidebarHeaderDropdownButton extends React.PureComponent {
                                 marginLeft: 4,
                             }}
-                            openModalOnClick={true}
+                            onClick={(e) => {
+                                e.stopPropagation();
+                                const customStatusInputModalData = {
+                                    ModalId: ModalIdentifiers.CUSTOM_STATUS,
+                                    dialogType: CustomStatusModal,
+                                    dialogProps: {userId: this.props.currentUser.id},
+                                };
+                                this.props.openModal(customStatusInputModalData);
+                            }}
```
nit - Create a named function.

manojmalik20 committed:
Modified the function for custom status emoji click in sidebar header
